.circleci/config.yml

```diff
@@ -145,34 +151,107 @@ jobs:
       - store_artifacts:
           path: test-results
 
-  unpackaged-sanity:
+  linux-tarball-smoke-tests:
+    <<: *defaults
+    parameters:
+      # If true, the tests will use the LTS version of node. If false, the latest version of node will be used instead.
+      lts:
+        type: boolean
+        default: true
+    steps:
+      - attach_workspace:
+          at: ~/repo
+
+      # Upgrade to whatever version of Node we need.
+      - node/install:
+          install-npm: false
+          lts: << parameters.lts >>
+
+      # Log the version of node we're using post-upgrade.
+      - run: node -v
+
+      - run:
+          name: "Install sfdx-cli"
+          command: npm install -g sfdx-cli
+
+      # Build and install our tarball.
+      - run:
+          name: "Build and install tarball"
+          command: |
+            yarn pack --filename tarball.tgz
+            # Pipe in a 'y' to simulate agreeing to install an unsigned plugin. Use a URI consisting of the file's full path.
+            echo y | sfdx plugins:install file://`pwd`/tarball.tgz
+
+      # Persist the workspace, so the tarball is visible to subsequent jobs in this flow.
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+
+      # Create a directory for the test output.
+      - run: mkdir test-results
+
+      # Run the smoke tests. If none of them explode, we can reasonably assume that the plugin is safe.
+      - run:
+          name: "Smoke Tests"
+          command: smoke-tests/smoke-test.sh sfdx
+
+      # Upload the results as an artifact instead of as test results, so they don't tank the build.
+      - store_artifacts:
+          path: test-results
+
+  windows-tarball-smoke-tests:
     executor:
-      name: win/default # executor type
+      name: win/default # Executor type
       size: "medium"
+    parameters:
+      # If true, the tests will use the LTS version of node. If false, the latest version of node will be used instead.
+      lts:
+        type: boolean
+        default: true
     working_directory: C:\repo
 
     steps:
-      # Commands are run in a Windows virtual machine environment
+      # Commands are run in a Windows virtual machine environment.
       - attach_workspace:
           at: C:\repo
 
+      # Upgrade node to whatever version we need.
       - run:
-          name: "Upgrade node to LTS"
+          name: "Upgrade node"
           shell: bash.exe
-          command: .circleci/windows-node-upgrade.sh lts
+          command: |
+            DESIRED_VERSION="current" && [[ << parameters.lts >> == true ]] && DESIRED_VERSION="lts"
```

Comment:
Can this use a consistent name with the upgrade script? The upgrade script uses `WANTED_VERSION`

Reply:
Will do.